### PR TITLE
feat(priority-alerts): Add a ratelimit to severity service calls

### PR DESCRIPTION
**Pull request by snigdhas**
Adding a ratelimit of 5 requests per second to the Seer severity service. We can adjust this if we see the ratelimit kick in as we're rolling out the service. 

Refactored a bit to only call the service for python/js projects and check the conditions before checking the ratelimit.

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -1829,13 +1830,10 @@ def _create_group(project: Project, event: Event, **group_creation_kwargs: Any)
     # add sdk tag to metadata
     group_data.setdefault("metadata", {}).update(sdk_metadata_from_event(event))
 
-    # Add severity to metadata for alert filtering for errors events.
-    # We can skip this if the group type is explicitly NOT an error.
+    # add severity to metadata for alert filtering
     group_type = group_creation_kwargs.get("type", None)
-    severity: Mapping[str, Any] = {}
-    if not group_type or group_type == ErrorGroupType.type_id:
```
Comment (scefali) [inline src/sentry/event_manager.py:1836]: @snigdhas should this code be removed?

Comment (snigdhas) [reply src/sentry/event_manager.py:1836]: yep, I've moved it into `_get_severity_metadata_for_group ` to keep severity-logic out of `_create_group `

**Review by scefali — APPROVED** [review]

File: src/sentry/event_manager.py [hunk 2]
```diff
@@ -2163,12 +2160,37 @@ def _process_existing_aggregate(
...
+    is_error_group = group_type == ErrorGroupType.type_id if group_type else True
+    feature_enabled = features.has("projects:first-event-severity-calculation", event.project)
+
+    should_calculate_severity = feature_enabled and is_supported_platform and is_error_group
+    if should_calculate_severity:
+        from sentry import ratelimits as ratelimiter
+
+        if ratelimiter.backend.is_limited(
+            f"seer:severity-calculation:{project_id}",
+            limit=5,
```
Comment (jangjodi) [inline src/sentry/event_manager.py:2188]: just a suggestion - you could make this an option so you can adjust it quickly once it's in production

**Comment by mrduncan** [comment]
Is there a global rate limit for this as well? A lot of projects making a couple of requests will add up fast

**Pull request merged**